### PR TITLE
KT-22142 Intentions: "Convert to primary constructor" changes semantics for property with custom setter

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-22142

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/intentions/ConvertSecondaryConstructorToPrimaryIntention.kt [hunk 1]
```diff
@@ -24,7 +24,17 @@ import org.jetbrains.kotlin.util.kind
 
 class ConvertSecondaryConstructorToPrimaryInspection : IntentionBasedInspection<KtSecondaryConstructor>(
     ConvertSecondaryConstructorToPrimaryIntention::class,
-    { constructor -> constructor.containingClass()?.secondaryConstructors?.size == 1 }
+    fun(constructor: KtSecondaryConstructor): Boolean {
+        val containingClass = constructor.containingClass() ?: return false
+        if (containingClass.secondaryConstructors.size != 1) return false
+        val parameterToPropertyMap = mutableMapOf<ValueParameterDescriptor, PropertyDescriptor>()
+        with(ConvertSecondaryConstructorToPrimaryIntention) {
+            constructor.extractInitializer(parameterToPropertyMap, containingClass.analyzeWithContent(), KtPsiFactory(constructor))
```
Comment (yanex) [inline idea/src/org/jetbrains/kotlin/idea/intentions/ConvertSecondaryConstructorToPrimaryIntention.kt]: It seems the result value of this expression is not used.

File: idea/src/org/jetbrains/kotlin/idea/intentions/ConvertSecondaryConstructorToPrimaryIntention.kt [hunk 2]
```diff
@@ -24,7 +24,17 @@ import org.jetbrains.kotlin.util.kind
...
+        }
+        return parameterToPropertyMap.none { (_, property) ->
+            (DescriptorToSourceUtils.descriptorToDeclaration(property) as? KtProperty)?.setter?.hasBody() == true
```
Comment (yanex) [inline idea/src/org/jetbrains/kotlin/idea/intentions/ConvertSecondaryConstructorToPrimaryIntention.kt]: Or a delegate, maybe?

**Comment by yanex** [comment]
Review 👍

**Comment by t-kameyama** [comment]
64f6ed586b6e8e836c98969a84d917e9ba576df7

**Pull request closed without merging**